### PR TITLE
Polish the auto TLS related docs

### PR DESCRIPTION
**Pull request by ZhiminXiang**
## Proposed Changes 

- Bump the min version of cert-manager to 1.0.0
- Remove the stale content used for installing cert-manager
- Move the part of configuring DNS challenge into auto TLS docs.

### CODE REVIEW

File: docs/serving/installing-cert-manager.md [hunk 1]
```diff
@@ -27,45 +27,14 @@ You must meet the following requirements to install cert-manager for Knative:
   component, see the [Knative installation guides](../install/).
 - You must configure your Knative cluster to use a
   [custom domain](./using-a-custom-domain.md).
-- Knative currently supports cert-manager version `0.12.0` or higher.
+- Knative currently supports cert-manager version `1.0.0` or higher.
```
[the remark this answers is not in the document]
Comment (ZhiminXiang) [reply docs/serving/installing-cert-manager.md:30]: release 0.19 started supporting it.

Comment (mpetason) [inline docs/serving/installing-cert-manager.md:30]:
```suggestion
- Knative currently supports cert-manager version `1.0.0` and higher.
```

File: docs/serving/using-auto-tls.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ You must meet the following prerequisites to enable auto TLS:
     [Contour, version 1.1 or higher](../install/any-kubernetes-cluster.md#installing-the-serving-component),
     or [Gloo, version 0.18.16 or higher](https://docs.solo.io/gloo/latest/installation/knative/).
     Note: Currently, [Ambassador](https://github.com/datawire/ambassador) is unsupported.
-  - [cert-manager version `0.12.0` or higher](./installing-cert-manager.md).
+- [cert-manager version `1.0.0` or higher](./installing-cert-manager.md).
```
Comment (mpetason) [inline docs/serving/using-auto-tls.md:47]:
```suggestion
- [cert-manager version `1.0.0` and higher](./installing-cert-manager.md).
```

File: docs/serving/using-auto-tls.md [hunk 3]
```diff
@@ -135,6 +135,20 @@ and which DNS provider validates those requests.
 
     Result: The `Status.Conditions` should include `Ready=True`.
 
+### DNS-01 challenge only: Configure your DNS provider
+
+If you choose to use DNS-01 challenge, configure which DNS provider is used to
+validate the DNS-01 challenge requests.
+
+Instructions about configuring cert-manager for any of the supported DNS
+providers are provided in
```
Comment (mpetason) [inline docs/serving/using-auto-tls.md:144]:
```suggestion
Instructions about configuring cert-manager, for all of the supported DNS
providers, are provided in
```

File: docs/serving/installing-cert-manager.md [hunk 4]
```diff
@@ -27,45 +27,14 @@ You must meet the following requirements to install cert-manager for Knative:
...
 
 ## Downloading and installing cert-manager
 
-Use the following steps to download, install, and configure cert-manager for
-your Knative cluster environment:
-
-1. Follow the steps in the official `cert-manager` website to download and install cert-manager
+Follow the steps in the official `cert-manager` website to download and install cert-manager
```
Comment (mpetason) [inline docs/serving/installing-cert-manager.md:34]:
```suggestion
Follow the steps from the official `cert-manager` website to download and install cert-manager
```

**Review by mpetason — COMMENTED** [review]
Small changes, content looks good!

File: docs/serving/installing-cert-manager.md (hunk 1 above)
Comment (ZhiminXiang) [reply docs/serving/installing-cert-manager.md]: Done

File: docs/serving/installing-cert-manager.md (hunk 4 above)
Comment (ZhiminXiang) [reply docs/serving/installing-cert-manager.md]: Done

File: docs/serving/using-auto-tls.md (hunk 3 above)
Comment (ZhiminXiang) [reply docs/serving/using-auto-tls.md]: Done

File: docs/serving/using-auto-tls.md (hunk 2 above)
Comment (ZhiminXiang) [reply docs/serving/using-auto-tls.md]: Done

**Review by nak3 — APPROVED** [review]
/lgtm

**Comment by ZhiminXiang** [comment]
@abrennan89 could you please help to review it? :)

**Review by tcnghia — APPROVED** [review]

**Pull request merged**